lib/src/model/opening_explorer/opening_explorer_repository.dart

```diff
@@ -0,0 +1,196 @@
+import 'package:dartchess/dartchess.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:http/http.dart';
+import 'package:lichess_mobile/src/constants.dart';
+import 'package:lichess_mobile/src/model/common/http.dart';
+import 'package:lichess_mobile/src/model/common/perf.dart';
+import 'package:lichess_mobile/src/model/opening_explorer/opening_explorer.dart';
+import 'package:lichess_mobile/src/model/opening_explorer/opening_explorer_preferences.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'opening_explorer_repository.g.dart';
+part 'opening_explorer_repository.freezed.dart';
+
+@riverpod
+Stream<OpeningExplorerEntry> openingExplorer(
```

Comment:
Thanks for the feedback, I can definitely see now why my cache implementation is subpar in many ways. One thing though, you say that a `StreamProvider` can be replaced by an `AsyncNotifier`, but I'm not really sure how that would be done. They yield keyword only works in combination with stream types, and using a regular return statement wouldn't work since we need to actually stream the results.

Reply:
You definitely can convert a StreamProvider to an AsyncNotifier. I've done it already in the app. I found that StreamProvider could lead to some subtle problems hard to see first, that's why I converted them.

An AsyncNotifier will produce state changes over time, that you can `watch`. So it does what a stream can do, except it is not a stream. To produce these state changes from a Stream you'd not use `for` and `yield` but `listen` to the stream inside the provider implementation.

Reply:
I've converted it now.